### ISSUE TITLE
error on Filtering on Nested Collections

### ISSUE THREAD

**Issue by masoodinfo** [alirezanet/Gridify#58]
hi tnx for add nested 
i use :
.AddMap("shomare",x=>x.RequestManager.select(y=>y.shomare))

.net 4.6
 i get error :
Cannot compare elements of type 'System.Collections.Generic.ICollection`1[[WebApp.Models.RequestManager, WebApp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]'. Only primitive types, enumeration types and entity types are supported

**Comment by alirezanet** [comment]
Hi,
What is the RequestManager class? 
you should add a reproducible example.

**Comment by masoodinfo** [comment]
It is a nested list(one to many realation)

**Comment by alirezanet** [comment]
These features have been tested in a lot of ways, but if you think this is a bug you should give me more information to help you.
Adding a producible example or exact code that caused this error could be handy.

Also, make sure to check out the docs [SubCollections](https://alirezanet.github.io/Gridify/guide/gridifyMapper.html#filtering-on-nested-collections).

[this test](https://github.com/alirezanet/Gridify/blob/610e5705eb32b72164234665b78b3c07520385fa/test/Gridify.Tests/GridifyNestedCollectionTests.cs#L107) could be useful 

اگر خواستید میتونید فارسی هم توضیح بدید مشکلی نیست

**Comment by masoodinfo** [comment]
سلام علی جان شرمنده 
.AddMap("prop1", l2 => l2.Level3List.Select(l3 => l3.Property1));
اگر Level3List نال باشه چی میشه ؟
من لیستم می تونه خالی باشه.خطا هم فکر میکنم سراین باشه.
راهی داره ؟
سپاس

**Comment by masoodinfo** [comment]
var map = new GridifyMapper<FaxRequest>()
                .AddMap("fname", x => x.FaxRequestFiles.Select(y => y.fname));
            var gridquery = new GridifyQuery(1, 100, "fname=*p1");
            var q = db.FaxRequests.Gridify(gridquery, map);

db is database context 
generated by edmx file 
دو تا جدول که ارتباط یک به چند با هم دارند.
یک درخواست فکس می تونه چند فایل داشته باشه
یک لیست دستی خودم ساختم جواب میده ولی از بانک رو جواب نمیگیرم
با تشکر

**Comment by alirezanet** [comment]
I think this is related to a limitation of old versions of EntityFramework, 
What EF version are you using? 

Can you please check this code works or not?
``` c#
db.FaxRequests.Where(q => q.FaxRequestFiles.Any(w=> w.fname.Contains("p1"))).ToList();
```

**Comment by masoodinfo** [comment]
![p1](https://user-images.githubusercontent.com/10396734/146164384-c85279ae-682c-485f-a846-1ee3fc1f9d34.png)
it work properly
EF  = 6.4.4

**Comment by alirezanet** [comment]
We don't have this problem with the recent versions of EF but thanks to you, I found the bug in EF6, I'll probably fix this for the next version.

thanks

**Comment by masoodinfo** [comment]
q.FaxRequestFiles != null

**Comment by masoodinfo** [comment]
بله درست میفرمایید
کد بالا رو تست کردم همون خطا رو میده
count میشه استفاده کرد انگار
به هرحال ممنونم ومنتظر

**Comment by alirezanet** [comment]
This is fixed in `v2.4.8`.
